src/details/ArborX_DetailsDistributor.hpp

```diff
@@ -270,10 +270,11 @@ class Distributor
                                   Kokkos::LayoutRight>));
 
     using MemorySpace = typename ExportView::memory_space;
-    static_assert(
-        std::is_same_v<MemorySpace, typename ImportView::memory_space>);
-    static_assert(
-        std::is_same_v<MemorySpace, typename decltype(_permute)::memory_space>);
+    static_assert(Kokkos::SpaceAccessibility<
+                  MemorySpace, typename ImportView::memory_space>::accessible);
+    static_assert(Kokkos::SpaceAccessibility<
+                  MemorySpace,
+                  typename decltype(_permute)::memory_space>::accessible);
```

Comment:
This would only allow `ImportView` to be in `SharedSpace`, right? Would it still be correct to allow both `Views` to be in a shared memory space?
Don't we care more about the spaces being accessible from the given execution space?